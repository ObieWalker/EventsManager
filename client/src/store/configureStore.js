import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers/rootReducer';

const middleware = applyMiddleware(thunkMiddleware, createLogger({ collapsed: true }));

const Store = createStore(
    rootReducer,
    applyMiddleware(middleware)
);

export default Store;