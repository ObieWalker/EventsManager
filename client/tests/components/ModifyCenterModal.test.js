import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import connect, { ModifyCenter }
  from '../../src/components/AdminDashboard/ModifyCenterModal';

let mountedComponent;
let props;
const mockStore = configureStore();
// const createCenter = jest.fn(() => Promise.resolve({}));

let wrapper;
let isCenterUpdating;
const updateSuccess = '';
const updateError = '';
const center = {
  name: 'random',
  address: 'random address',
  city: 'random',
  capacity: 500,
  facility: 'Tables and Chairs'
};
const modifyCenter = jest.fn(() => Promise.resolve({}));
const getComponent = () => {
  if (!mountedComponent) {
    props = {
      center,
      modifyCenter,
      isCenterUpdating,
      updateSuccess,
      updateError
    };
    mountedComponent = shallow(<ModifyCenter {...props} />);
  }
  return mountedComponent;
};

describe('Modify Center Component', () => {
  beforeEach(() => {});
  wrapper = getComponent();

  it('component successfully rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().reset).toBeDefined();
  });
  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().handleChange).toBeDefined();
  });
  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().isValid).toBeDefined();
  });
  it('should have a method that handles change to each input element', () => {
    expect(wrapper.instance().updateCenter).toBeDefined();
  });
  it('testing create center function', () => {
    wrapper = getComponent();
    const e = { preventDefault: () => undefined };
    e.preventDefault = jest.fn();
    wrapper.instance().updateCenter(e);
    expect(e.preventDefault).toHaveBeenCalled();
  });
//   it('testing clear function', () => {
//     wrapper = getComponent();
//     wrapper.instance().reset();
//     expect(wrapper.state().name).toEqual('');
//     expect(wrapper.state().address).toEqual('');
//     expect(wrapper.state().city).toEqual('');
//     expect(wrapper.state().capacity).toEqual('');
//     expect(wrapper.state().facility).toEqual('');
//     expect(wrapper.state().errors).toEqual({});
//   });
});

describe('Connected Component', () => {
  describe('Connected mountedLogin', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        isCenterUpdating,
        updateSuccess,
        updateError,
        center,
        modifyCenter
      });
      wrapper = shallow(<connect store={store}
        isCenterUpdating={isCenterUpdating}
        updateSuccess={updateSuccess}
        updateError={updateError}
        center={center}
        modifyCenter={modifyCenter}
      />);
      expect(wrapper.length).toBe(1);
    });
  });
});
