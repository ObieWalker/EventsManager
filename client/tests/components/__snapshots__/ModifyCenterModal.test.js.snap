// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modify Center Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModifyCenter
    center={
      Object {
        "address": "random address",
        "capacity": 500,
        "city": "random",
        "facility": "Tables and Chairs",
        "name": "random",
      }
    }
    modifyCenter={[MockFunction]}
    updateError=""
    updateSuccess=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <CenterForm
        center={
          Object {
            "address": "random address",
            "capacity": 500,
            "city": "random",
            "facility": "Tables and Chairs",
            "name": "random",
          }
        }
        defaultAddressValue="random address"
        defaultCapacityValue={500}
        defaultCenterName="random"
        defaultCityValue="random"
        defaultFacilityValue="Tables and Chairs"
        errors={Object {}}
        modifyOnClick={[Function]}
        onChange={[Function]}
      />,
      "className": "modifyCenterForm",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "addressValue": undefined,
        "capacityValue": undefined,
        "center": Object {
          "address": "random address",
          "capacity": 500,
          "city": "random",
          "facility": "Tables and Chairs",
          "name": "random",
        },
        "centerNameValue": undefined,
        "cityValue": undefined,
        "defaultAddressValue": "random address",
        "defaultCapacityValue": 500,
        "defaultCenterName": "random",
        "defaultCityValue": "random",
        "defaultFacilityValue": "Tables and Chairs",
        "errors": Object {},
        "facilityValue": undefined,
        "modifyOnClick": [Function],
        "onChange": [Function],
        "onFocus": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <CenterForm
          center={
            Object {
              "address": "random address",
              "capacity": 500,
              "city": "random",
              "facility": "Tables and Chairs",
              "name": "random",
            }
          }
          defaultAddressValue="random address"
          defaultCapacityValue={500}
          defaultCenterName="random"
          defaultCityValue="random"
          defaultFacilityValue="Tables and Chairs"
          errors={Object {}}
          modifyOnClick={[Function]}
          onChange={[Function]}
        />,
        "className": "modifyCenterForm",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addressValue": undefined,
          "capacityValue": undefined,
          "center": Object {
            "address": "random address",
            "capacity": 500,
            "city": "random",
            "facility": "Tables and Chairs",
            "name": "random",
          },
          "centerNameValue": undefined,
          "cityValue": undefined,
          "defaultAddressValue": "random address",
          "defaultCapacityValue": 500,
          "defaultCenterName": "random",
          "defaultCityValue": "random",
          "defaultFacilityValue": "Tables and Chairs",
          "errors": Object {},
          "facilityValue": undefined,
          "modifyOnClick": [Function],
          "onChange": [Function],
          "onFocus": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
